services/tilesets.js

```diff
@@ -33,13 +39,45 @@ var Tilesets = {};
  */
 Tilesets.listTilesets = function(config) {
   v.assertShape({
-    ownerId: v.string
+    ownerId: v.required(v.string),
+    limit: v.range([1, 500]),
+    sortBy: v.oneOf('created', 'modified'),
+    start: v.string,
+    type: v.oneOf('raster', 'vector'),
+    visibility: v.oneOf('public', 'private')
   })(config);
 
   return this.client.createRequest({
     method: 'GET',
     path: '/tilesets/v1/:ownerId',
-    params: config
+    params: pick(config || {}, ['ownerId']),
```

Comment:
This isn't right! Or the line below!

If I don't have a default object then I get a lot of failing tests which seem to be to do with pagination. I can't see what I'm doing differently here when compared to something like the datasets service.

Reply:
Oh hm, this is kind of a weird one because there are no *required* query params. There are a couple of things we could do: 

1. `ownerId` is required, so we'll want to update the `'works'` test to include it (I think this changed since the service was initially added to `sdk-js`).
2. Since there are no required (or even publicly documented 😂) query params for the list tilesets endpoint, there do seem to be legitimate situations where there will be no `query`s to `pick`. Your approach seems pretty solid to me, but I might change it to something like `config ? pick(config, ['blah', 'bluh']) : {}` to make it a little clearer what's happening. Mostly personal preference, but I think this would also prevent some unnecessary `pick` calls.

Reply:
Makes sense. I've updated the `works` test and the picking of config params. I went with the ternary as I agree that it looks a little clearer than having the logic inside the function parameters.

Reply:
@danswick Is ownerId required in this method? https://github.com/mapbox/mapbox-sdk-js/blob/master/lib/classes/mapi-request.js#L127 set's it based on the access token if not supplied.